docs/cookbook/languages/python.md

```diff
@@ -0,0 +1,289 @@
+  $ flox install <package>   <- Install a package into an environment
+  $ flox activate            <- Enter the environment
+  $ flox edit                <- Add environment variables and shell hooks
+```
+
+This Flox environment is now ready to be populated with packages.
+
+### Select a Python interpreter
+
+To begin, we need a Python interpreter. For this example, we will be using Python 3.11. Search for the version of Python your project requires, omitting the dot between major and minor numbers:
+
+```
+flox search python311
+```
+
+This will show you the list of packages matching the major and minor version of Python you specified:
+
+```
+$ flox search python311
+python311      High-level dynamically-typed programming language
+python311Full  High-level dynamically-typed programming language
+
+Use 'flox show <package>' to see available versions
+```
+
+To see the specific versions of Python available in the Flox Catalog, use the `flox show` command:
+
+```
+flox show python311Full
+```
+
+This will list all of the releases that match the major and minor numbers you provided:
+
+```
+$ flox show python311Full
+python311Full - High-level dynamically-typed programming language
+    python311Full@python3-3.11.9
+    python311Full@python3-3.11.8
+    python311Full@python3-3.11.7
+    python311Full@python3-3.11.6
+    python311Full@python3-3.11.5
+    python311Full@python3-3.11.4
+    python311Full@python3-3.11.3
+    python311Full@python3-3.11.2
+    python311Full@python3-3.11.1
+```
+
+At this point, you can install the latest version by running `flox install` without a version specified:
+
+```
+flox install python311Full
+```
+
+This will show a message indicating the package was successfully installed:
+
+```
+$ flox install python311Full
+✅ 'python311Full' installed to environment 'new-python-project'
+$ flox list
+python311Full: python311Full (python3-3.11.9)
+```
+
+When `flox upgrade` is run in this environment, the version of Python will be upgraded to the latest version available in the catalog. If you wish to pin your Flox environment to a specific version of Python, you can specify it in the `flox install` command:
+
+```
+flox install python311Full@python3-3.11.3
+```
+
+### Add Python packages
+
+Python projects often require a collection of packages in addition to an interpreter. Often these are installed using `pip`, but they can also be acquired from the Flox Catalog. This will allow them to be locked in the Flox Manifest alongside the other packages in your environment.
+
+To find these Python packages in the Flox Catalog, use the same `flox search` syntax from before:
+
+```
+flox search numpy
+```
+
+This will return the set of packages that match your search term:
+
+```
+$ flox search numpy
+[...]
+python311Packages.numpy    Scientific tools for Python
+[...]
+```
+
+Python packages will have a prefix of `pythonXXXPackages.` in their package name. To install them, use the entire package name including the prefix:
```

Comment:
Probably worth mentioning that the version needs to match the version of the interpreter or they won't be found at run time.